packages/lib-classifier/src/hooks/useTextData.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+
+function getSubjectUrl(subject) {
+  // Find the first location that has a text MIME type.
+  const textLocation = subject.locations.find(l => l['text/plain']) || {}
+  const url = Object.values(textLocation)[0]
+  if (url) {
+    return url
+  } else {
+    throw new Error('No text url found for this subject')
+  }
+}
+
+async function requestData(subject) {
+  const url = getSubjectUrl(subject)
+  const response = await fetch(url)
+  if (!response.ok) {
+    const error = new Error(response.statusText)
+    error.status = response.status
+    throw error
+  }
+  const body = await response.text()
+  return body
+}
+
+/**
+  A custom hook to load text data subjects from Panoptes
+*/
+export default function useTextData(
+  /** Panoptes subject */
+  subject,
+  /** on data ready callback */
+  onReady,
+  /** on error callback */
+  onError
+) {
+  const [textData, setTextData] = useState('')
+
+  useEffect(() => onSubjectChange(), [subject])
```

Comment:
Best practice in React would be to define any functions inside the effect hook. Otherwise, you should add `onSubjectChange` as an external dependency.
https://beta.reactjs.org/reference/react/useEffect#specifying-reactive-dependencies

Reply:
`onReady` and `onError` should also be dependencies here, since they are called inside the body of the hook.